engine/src/main/java/org/hibernate/validator/internal/metadata/aggregated/BeanMetaDataImpl.java

```diff
@@ -280,8 +306,24 @@ public PropertyMetaData getMetaDataFor(String propertyName) {
 	}
 
 	@Override
-	public ExecutableMetaData getMetaDataFor(Executable executable) {
-		return executableMetaDataMap.get( ExecutableHelper.getSignature( executable ) );
+	public Optional<ExecutableMetaData> getMetaDataFor(Executable executable) {
+		String signature = ExecutableHelper.getSignature( executable );
```

Comment:
First I thought about having 2 separate methods but the call to `ExecutableHelper.getSignature( executable )` has a higher cost than expected. Thus, it's better to have only one method.

I think the `Optional` makes it clear that the behavior is a bit different. We might even rename the method to differentiate it better? Maybe `getConstraintMetaDataFor()`?